File: android/app/src/main/java/jagtheora/theora/TheoraInfo.java
package jagtheora.theora;

import jagtheora.misc.SimplePeer;

public final class TheoraInfo extends SimplePeer {

    public int frameWidth;

    public int picX;

    public int pixelFormat;

    public int picHeight;

    public int frameHeight;

    public int fpsDenominator;

    public int picY;

    public byte versionSubMinor;

    public int fpsNumerator;

    public int aspectDenominator;

    public int colourSpace;

    public byte versionMinor;

    public byte versionMajor;

    public int picWidth;

    public int aspectNumerator;

    static {
        initFields();
    }

    public static void initFields() {}

    public TheoraInfo() {
        this.init();
        if (this.b()) {
            throw new IllegalStateException();
        }
    }

    protected void init() {}

    protected void clear() {}
}

File: src/commonMain/kotlin/jagtheora/theora/TheoraInfo.kt
package jagtheora.theora

import jagtheora.misc.SimplePeer

expect class TheoraInfo() : SimplePeer {
    val colourSpace: Int

    val versionSubMinor: Byte

    var fpsNumerator: Int

    val pixelFormat: Int

    val picY: Int

    val aspectDenominator: Int

    val picWidth: Int

    val versionMinor: Byte

    val picX: Int

    val picHeight: Int


    var fpsDenominator: Int

    val versionMajor: Byte


    var frameWidth: Int

    val aspectNumerator: Int


    var frameHeight: Int

    override fun clear()

    companion object {

        /*private*/ fun initFields()

        /*protected*/ fun init()
    }
}
